### ISSUE TITLE
Upgrade GitHub CI/CD Workflows to Current Stable Version

### ISSUE THREAD

**Issue by aj-stein-nist** [usnistgov/oscal-content#116]
# User Story:

As an OSCAL tool developer, in order to make sure CI/CD is properly developed and integrated in relevant OSCAL projects, I want the current version of GitHub Actions CI/CD workflows updated to current stable.

## Goals:

The NIST OSCAL Team will complete cleaning, deduplification, and misc improvements to the GitHub Actions in [usnistgov/OSCAL](https://github.com/usnistgov/OSCAL/) for improved maintenance, stability, and resiliency.

## Dependencies:

- [x] Completion of usnistgov/OSCAL#1223
- [x] usnistgov/metaschema#235
- [x] usnistgov/OSCAL#1698

## Acceptance Criteria

- [x] All readme documentation affected by the changes in this issue have been updated.
- [x] A Pull Request (PR) is submitted that fully addresses the goals of this User Story. This issue is referenced in the PR.
- [x] The CI-CD build process runs without any reported errors on the PR. This can be confirmed by reviewing that all checks have passed in the PR.

**Comment by aj-stein-nist** [comment]
Earlier in the week I spent I looked deeply into this and noticed that very little of the updated between the main OSCAL CI/CD and oscal-content because the different outcomes desired from the repository (oscal-content only converts content). If anything, this means maybe only the steps for setting up Java and Saxon and/or Calabash matter here. That means at this point simple copy-paste.

I did some experiment with cleanup of this and had a think, or quick diversionary spike to think of how we can finish the modularization here better than copy-pasting, and that would mean beginning the move to reusable workflows. The alternative, composite actions, are not able to access tokens, so flat out could not commit back content (this is a very important stumbling block and the final step in pipeline processing for this repo). So here are the spike experiments.

https://github.com/aj-stein-nist/test-gha-reusable-workflows/tree/main/.github/workflows

I am worried because of token management from forks and other things from testing this likely will not work particularly well is token management around this committing documents issue. Tokens are determined in the checkout step, in the default action provided by the GHA team. I am struggling to get around https://github.com/actions/checkout/issues/298 which means this spike might be wrapping up with a "meh, this is not going to work either" result.

**Comment by aj-stein-nist** [comment]
Ok, #1470 created to tie off the spike work and return back to the AC here of just upgrading things to a stable version. New PR incoming.

**Comment by david-waltermire-nist** [comment]
@aj-stein-nist I want to keep this open until PR 157 is resolved.

**Comment by aj-stein-nist** [comment]
Moving to Sprint 61.

**Comment by aj-stein-nist** [comment]
Need to move to sprint 63. Much headway was made but ore needed to wrap this up with testing metaschema submodule of OSCAL and OSCAL submodul of oscal-content.

**Comment by aj-stein-nist** [comment]
Testing this again in https://github.com/usnistgov/oscal-content/pull/183. I accidentally closed an older branch and messed it up.

**Comment by aj-stein-nist** [comment]
Still working through usnistgov/metaschema#235 which is breaking builds. I am primarily focused on that for now to further debug, fix, and unblock this work. More to follow next week.

**Comment by aj-stein-nist** [comment]
I had not yet updated this before the conclusion of Sprint 63. I needed to move this issue to Sprint 64.

I was able to completely validate testhttps://github.com/usnistgov/oscal-content/pull/183 to get end to end testing of touched-up changes needed in develop with #235. This particular issue is blocked until we can merge #235 changes (updated unit tests pending) *and*  https://github.com/usnistgov/OSCAL/pull/1639 changes when they are cherry-picked and re-adapted. The latter needs to be done after testing the commit because that PR, per community developer galtm, is connected to feature work that is not specific to the bug fix.

I expect to work towards resolution on both of these and unblock the latter before the end of Sprint 64.

**Comment by aj-stein-nist** [comment]
I will be attempting to complete work on usnistgov/metaschema#235 this evening with tests but this will still need to be reviewed and merged by the Metaschema team before this code can be worked.

If others wish to pick it up before end of sprint they can let me know, but totally understandable to leave it as-is. I will pick it up if so when I am back (metaschema or CI/CD work, in that order).

**Comment by aj-stein-nist** [comment]
I had to take the time to review usnistgov/metaschema#235's PR and will work towards final review after testing and final review, then integrate it up the change to resolve the CI/CD issues. I will mark as blocked usnistgov/OSCAL#1698 will inevitably block a lot of CI/CD working if it is wrapped in after merged and approved. I was unable to troubleshoot it correctly after 1.5 hours of trying yesterday, so usnistgov/OSCAL#1698 is now pulled into the sprint.

**Comment by aj-stein-nist** [comment]
It seems I have been working on #116 more or less the whole of last sprint and linking working to the PR not knowing this issue has been sitting dormant, I guess pull this in and close with the associated PR to make things more accurate.

**Comment by aj-stein-nist** [comment]
Moving to Sprint 77 as it is incomplete.

**Comment by aj-stein-nist** [comment]
The code has been merged and will be prepared for release.
